### ISSUE TITLE
CORS error when installing extensions in Cursor.sh using devcontainers

### ISSUE THREAD

**Issue by vasco** [getcursor/cursor#1477]
I'm encountering a problem while setting up a Python project in Cursor.sh using devcontainers. During the extension installation process, Cursor.sh gets stalled and doesn't proceed further. The issue seems to be related to a Cross-Origin Resource Sharing (CORS) error.
**Steps to Reproduce:**

1. Set up a Python project in Cursor.sh using devcontainers.
1. Configure the devcontainer.json file with the necessary extensions and settings.
1. Start the devcontainer build process.
1. Observe that Cursor.sh gets stalled during the extension installation step.

**Expected Behavior:**
Cursor.sh should successfully install all the specified extensions and complete the devcontainer setup process without any CORS-related errors.

**Actual Behavior:**
Cursor.sh encounters a CORS error during the extension installation process and gets stalled indefinitely. The developer tools show the following error messages:

```
Access to XMLHttpRequest at 'https://marketplace.cursorapi.com/_apis/public/gallery/extensionquery' from origin 'vscode-file://vscode-app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Failed to load resource: https://marketplace.cursorapi.com/_apis/public/gallery/extensionquery:1 net::ERR_FAILED
```

**Screenshots:**
![CleanShot 2024-06-03 at 18 25 01](https://github.com/getcursor/cursor/assets/87291/c7ee1294-d03b-4dc6-b5fd-d7d76e591136)

**Operating System**

* Version: 0.34.5
* VSCode Version: 1.89.1
* Commit: 1aa253fa263df64530be0cab8b4e791d45a1a820
* Date: 2024-06-02T18:36:22.954Z
* Electron: 28.2.8
* ElectronBuildId: undefined
* Chromium: 120.0.6099.291
* Node.js: 18.18.2
* V8: 12.0.267.19-electron.0
* OS: Darwin arm64 22.6.0

** Additional Information:** 

* I'm using the latest version of Cursor.sh.
* I've tried rebuilding the container with the --no-cache flag, but the issue persists.
* I've ensured that I have a stable internet connection during the container build process.
* I've checked the Cursor.sh documentation and community forums but couldn't find a specific solution for this issue.

I suspect that the problem is related to the CORS configuration on the https://marketplace.cursorapi.com server. It seems that the server is not sending the necessary Access-Control-Allow-Origin header in the response, causing the browser to block the request.

I would appreciate any guidance or assistance in resolving this CORS issue and enabling smooth extension installation in Cursor.sh using devcontainers.

**Comment by leerob** [comment]
Thanks for opening this issue! We've moved our community support to the [Cursor Forum](https://forum.cursor.com) where you can get help from both the team and other users.

If this is a specific bug that you can still reproduce, please open a new issue with clear reproduction steps on the forum.

Thanks you!
